services/packages/maven/cleanup.go

```diff
@@ -0,0 +1,112 @@
+package maven
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/gitea/models/packages"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/packages/maven"
+	"code.gitea.io/gitea/modules/setting"
+	packages_service "code.gitea.io/gitea/services/packages"
+)
+
+// CleanupSnapshotVersion removes outdated files for SNAPHOT versions for all Maven packages.
+func CleanupSnapshotVersions(ctx context.Context) error {
+	retainBuilds := setting.Packages.RetainMavenSnapshotBuilds
+	log.Info("Starting CleanupSnapshotVersion with retainBuilds: %d", retainBuilds)
```

Comment:
Again, I think it's a little too much logs by default.
Perhaps degrade them to `debug`?